lib/lib-shared-models/src/main/java/org/sagebionetworks/repo/model/AccessRecordExtractorUtil.java

```diff
@@ -0,0 +1,54 @@
+package org.sagebionetworks.repo.model;
+
+import java.lang.reflect.Field;
+
+/**
+ * Extract the ID and concreteType from objects.
+ *
+ * @author John
+ */
+public class AccessRecordExtractorUtil {
+    /**
+     * Exact the ID from an object
+     *
+     * @param object
+     * @return
+     */
+    public static String getObjectId(Object object) {
+        if (object == null) return null;
+        try {
+            Field field = object.getClass().getDeclaredField("id");
+            field.setAccessible(true);
+            Object ob = field.get(object);
+            if (ob instanceof String) {
+                return (String) ob;
+            } else if (ob instanceof Long) {
+                return ((Long) ob).toString();
+            } else return null;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * Exact the concrete type from an object
+     *
+     * @param object
+     * @return
+     */
+    public static String getConcreteType(Object object) {
```

Comment:
No duplicate code in production. Also this implements logic that is not needed (e.g. instance of Long)